kms, vision: tests should use t.SystemTest

```
2020/04/24 13:04:21 failed to set up kms tests - missing GOLANG_SAMPLES_PROJECT_ID?
FAIL	github.com/GoogleCloudPlatform/golang-samples/kms	0.082s
```

another one:

```
--- FAIL: TestSetEndpoint (1.37s)
    set_endpoint_test.go:51: DetectTexts: rpc error: code = PermissionDenied desc = Your application has authenticated using end user credentials from the Google Cloud SDK or Google Cloud Shell which are not supported by the vision.googleapis.com. We recommend configuring the billing/quota_project setting in gcloud or using a service account through the auth/impersonate_service_account setting. For more information about service accounts and how to use them in your application, see https://cloud.google.com/docs/authentication/.
FAIL
FAIL	github.com/GoogleCloudPlatform/golang-samples/vision/detect	1.403s
```

--- a/kms/kms_test.go
+++ b/kms/kms_test.go
@@ -40,7 +40,8 @@ var fixture *kmsFixture
 func TestMain(m *testing.M) {
 	tc, ok := testutil.ContextMain(m)
 	if !ok {
-		log.Fatal("failed to set up kms tests - missing GOLANG_SAMPLES_PROJECT_ID?")
+		log.Print("skipping - unset GOLANG_SAMPLES_PROJECT_ID?")
+		return
 	}
 
 	var err error

